GeneticDFA/DFAGeneticAlgorithm.cs

```diff
@@ -89,14 +100,78 @@ public void Start()
     // Needs proper implementation
     private bool EvolveOneGeneration()
     {
-        throw new NotImplementedException();
-
-        /*
-        IList<IChromosome> parents = SelectParents();
-        IList<IChromosome> chromosomeList = Cross(parents);
-        Mutate(chromosomeList);
-        Population.CreateNewGeneration(Reinsert(chromosomeList, parents));
-        return EndCurrentGeneration(); */
+        // First use Elitist selection to include elites
+        IList<IChromosome> selectedElites =
+            Selection.SelectChromosomes(SelectionScale(), Population.CurrentGeneration);
+        // Use roulette wheel selection on the selected elites. This can cause elites to be mutated multiple times
+        IList<IChromosome> selectedForModification = DFARouletteWheelSelection.SelectChromosomes(Population.MaxSize, selectedElites);
+        IList<IChromosome> newPopulation = new List<IChromosome>();
+
+        try
+        {
+            for (int i = 0; i < selectedForModification.Count; i++)
+            {
+                if (_rnd.GetDouble(0, 1) < MutationProbability)
+                {
+                    int i1 = i;
```

Comment:
hvad formål har int i1=i her?
den blev brugt før men ikke længere?

Reply:
Det er for at undgå at threadsene bruger i, som jo er en shared variable

Reply:
ah oright, smart smart.